core/serialize.cpp

```diff
@@ -1127,7 +1127,9 @@ bool dc_serialize(void **data, unsigned int *total_size)
 	LIBRETRO_S(reg_dimm_status);
 	LIBRETRO_S(NaomiDataRead);
 
-	LIBRETRO_S(cycle_counter);
+	#if FEAT_SHREC != DYNAREC_NONE
```

Comment:
This makes the save state format differ when using a dynarec and when not using one. Here the `cycle_counter` doesn't need to be de/serialized anyway so just do:
```
    LIBRETRO_S(i); // cycle_counter
```